src/deadline/cinema4d_submitter/font_installer.py

```
@@ -0,0 +1,286 @@
+    """
+    Uninstall provided font from the worker machine
+
+    :param src_path: path of font that needs to be removed
+
+    :returns: boolean that represents if the font was uninstalled and a string with any traceback that was created
+    """
+    if sys.platform != "win32":
+        return False, "Font uninstallation is only supported on Windows"
+
+    try:
+        # Determine where the font was installed
+        if scope == INSTALL_SCOPE_SYSTEM:
+            dst_path = os.path.join(FONT_LOCATION_SYSTEM, os.path.basename(src_path))
+            registry_scope = winreg.HKEY_LOCAL_MACHINE
+        else:
+            dst_path = os.path.join(FONT_LOCATION_USER, os.path.basename(src_path))
+            registry_scope = winreg.HKEY_CURRENT_USER
+
+        # Remove the fontname/filename from the registry
+        fontname = get_font_name(dst_path)
+
+        with winreg.OpenKey(registry_scope, FONTS_REG_PATH, 0, access=winreg.KEY_SET_VALUE) as key:
+            winreg.DeleteValue(key, fontname)
+
+        # Unload the font in the current session
+        if not gdi32.RemoveFontResourceW(dst_path):
+            os.remove(dst_path)
+            raise OSError(f'RemoveFontResourceW failed to load "{src_path}"')
+
+        if os.path.exists(dst_path):
+            os.remove(dst_path)
+
+        # Notify running programs
+        user32.SendMessageTimeoutW(
+            HWND_BROADCAST, WM_FONTCHANGE, 0, 0, SMTO_ABORTIFHUNG, 1000, None
+        )
+    except Exception:
+        return False, traceback.format_exc()
+    return True, ""
+
+
+def _install_fonts(session_dir: str) -> None:
+    """
+    Calls all needed functions for installing fonts
+
+    :param session_dir: directory of the session
+    """
+    logger.info("Looking for fonts to install...")
+    fonts = find_fonts(session_dir)
+
+    if not fonts:
+        logger.info("No custom fonts found, continuing task...")
+        return
```

Do we need this log line? If not, we can drop the block

---

> Do we need this log line?

I think the log line is useful if for some reason the worker is unable to find fonts. But I wonder if we should just raise an error instead because a job environment should not have been added to install fonts if there was no fonts to install.